src/collections/Menu/Menu.js

```diff
@@ -128,6 +145,38 @@ class Menu extends Component {
   static Item = MenuItem
   static Menu = MenuMenu
 
+  renderItems() {
+    const { items, onItemClick } = this.props
+
+    this.trySetState({ activeIndex: _.findIndex(items, ['active', false]) || -1 })
+
+    return _.map(items, (index, item) => {
+      const { content, childKey, name, onClick, itemProps } = item
+
+      const active = this.state.activeIndex === index
+      const finalKey = childKey || [content, name].join('-')
+
+      const handleClick = (e, { itemName, itemIndex }) => {
+        const handler = onItemClick || onClick
```

Comment:
If the user passes both a Menu `onItemClick` and an Item `onClick`, their onClick will never be called.  Instead of choosing one or the other, I think we should call each separately:

```js
if (onClick) onClick(e, { itemName, itemIndex })
if (onItemClick) onItemClick(e, { itemName, itemIndex })
```

I've also called them in order here, from the Item up to the Menu.